smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/retries/StandardRetryIntegration.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.kotlin.codegen.retries
+
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.codegen.core.SymbolReference.*
+import software.amazon.smithy.kotlin.codegen.core.*
+import software.amazon.smithy.kotlin.codegen.integration.KotlinIntegration
+import software.amazon.smithy.kotlin.codegen.model.buildSymbol
+import software.amazon.smithy.kotlin.codegen.model.namespace
+import software.amazon.smithy.kotlin.codegen.rendering.ClientConfigProperty
+import software.amazon.smithy.kotlin.codegen.rendering.protocol.HttpFeatureMiddleware
+import software.amazon.smithy.kotlin.codegen.rendering.protocol.ProtocolGenerator
+import software.amazon.smithy.kotlin.codegen.rendering.protocol.ProtocolMiddleware
+
+/**
+ * Adds retry wrappers around operation invocations, driven by client config (which specifies the [RetryStrategy]) and
+ * a middleware (which specifies the [RetryPolicy][aws.smithy.kotlin.runtime.retries.RetryPolicy]). This class is
+ * `open` and may be subclassed to customize the strategy and/or policy. Note that codegen which specifies a subclass of
+ * this should probably configure [replacesIntegrations] to remove the base implementation—otherwise, multiple copies of
+ * the config properties may be included in client config.
+ */
+open class StandardRetryIntegration : KotlinIntegration {
+    override fun additionalServiceConfigProps(ctx: CodegenContext): List<ClientConfigProperty> =
+        listOf(
+            ClientConfigProperty {
+                val retryStrategyBlock = """
+                    run {
+                        val strategyOptions = StandardRetryStrategyOptions.Default
+                        val tokenBucket = StandardRetryTokenBucket(StandardRetryTokenBucketOptions.Default)
+                        val delayer = ExponentialBackoffWithJitter(ExponentialBackoffWithJitterOptions.Default)
+                        StandardRetryStrategy(strategyOptions, tokenBucket, delayer)
+                    }
+                """.trimIndent()
+
+                symbol = buildSymbol {
+                    name = "RetryStrategy"
+                    namespace(KotlinDependency.CORE, "retries")
+                    nullable = false
```

Comment:
question

This implies we wouldn't be able to set the strategy to `null` someday correct? To get "no retry" behavior someone would have to explicitly pass `retryStrategy = RetryStrategy.None` or something. I think that's probably ok (and more clear) but just clarifying our stance.